### ISSUE TITLE
Improve the model

### ISSUE THREAD

**Issue by lindouer** [DL4mHealth/Medformer#3]
Hi, what changes can I make to the model to improve performance?

**Comment by lindouer** [comment]
What adjustments do I need to make to Medfomer if I want to change to Transfomer designed specifically for ECG data?

**Comment by YiheWang** [comment]
Hi, if you are asking for parameter tuning to improve performance, you can adjust the list of patch length `--patch_len_list` or change the augmentation method `--augmentations` . Other not method-specific parameters such as `--e_layers`, `--batch_size`, `--d_model`, `--d_ff`, `--learning_rate`,` --train_epochs`, `--patience` are also important to the result. For a fair comparison with baselines, we did not tune them, but you can tune them during real-world application.

If you want to modify the model structure specifically for ECG data, the easiest way is to find some ECG-specific augmentation methods and combine them with our method. If you pursue more modification, consider utilizing the domain knowledge of ECG. How are they different from other medical time series data? 

Besides, considering the abundance of public ECG data, applying self-supervised pretraining and using our method as a backbone might also be a good research direction.

**Comment by YiheWang** [comment]
```
for padding, value_embedding in zip(self.paddings, self.value_embeddings):
    x_new = padding(x).unsqueeze(1)  # (batch_size, 1, enc_in, seq_len+stride)
    x_new = value_embedding(x_new)  # (batch_size, d_model, patch_num, 1)
    x_new = x_new.squeeze().transpose(1, 2)  # (batch_size, patch_num, d_model)
    # Per patch augmentation
    aug_idx = random.randint(0, len(self.augmentation) - 1)
    x_new = self.augmentation[aug_idx](x_new)
    x_list.append(x_new)
```
Moreover, you can change how augmentations are applied during data embedding in Embed.py. Now we first embed the data and then apply data augmentations to the data embedding, following the famous work TS2Vec. According to more experiments after finishing our paper, we find applying augmentations to the raw data might improve the results and adapt to a greater variety of data augmentation methods, such as temporal flipping and channel shuffling.

**Comment by lindouer** [comment]
Thank you very much for your guidance
